pinot/generative/semi_supervised_net.py

```diff
@@ -106,17 +91,22 @@ def loss(self, g, y):
             # Only compute supervised loss for the labeled data
             h_not_none = h[~torch.isnan(y).flatten(), :]
             y_not_none = y[~torch.isnan(y)].unsqueeze(1)
-            # Convert to cuda if available
-            y_not_none = y_not_none.to(self.device)
+            y_not_none = y_not_none
             # The output-regressor needs to implement a loss function
             supervised_loss = self.loss_supervised(h_not_none, y_not_none)
 
         total_loss = supervised_loss.sum() + unsup_loss*self.unsup_scale
         return total_loss
 
     def loss_supervised(self, h, y):
-        # return self.output_regressor.loss(h, y)
-        return self._loss(h, y)
+        # If output regressor has loss function implemented
```

Comment:
hey if you call _loss it's already taken care of

Reply:
oh yeah, youre right